src/query-builder/SelectQueryBuilder.ts

```diff
@@ -117,35 +117,35 @@ export class SelectQueryBuilder<Entity> extends QueryBuilder<Entity> {
     /**
      * Adds new selection to the SELECT query.
      */
-    addSelect(selection: (qb: SelectQueryBuilder<any>) => SelectQueryBuilder<any>, selectionAliasName?: string): this;
+    addSelect(selection: (qb: SelectQueryBuilder<any>) => SelectQueryBuilder<any>, selectionAliasName?: string, virtual?: boolean): this;
```

Comment:
"virtual" is internal. We should not expose this functionality to the user. Once you move logic about into query builder you won't need to change this method at all